perfkitbenchmarker/spark_service.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Benchmarking support for Apache Spark services.
+
+In order to benchmark Apache Spark services such as Google Cloud
+Platform's Dataproc or Amazon's EMR, we create a BaseSparkService
+class.  Classes to wrap each provider's Apache Spark Service are
+in the provider directory as a subclass of BaseSparkService.
+
+Also in this module is a PkbSparkService, which builds a Spark
+cluster by creating VMs and installing the necessary software.
+
+For more on Apache Spark: http://spark.apache.org/
+"""
+
+import abc
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import resource
+
+FLAGS = flags.FLAGS
+
+# Cloud to use for pkb-created Spark service.
+PKB_MANAGED = 'pkb_managed'
+PROVIDER_MANAGED = 'managed'
+
+# This is used for error messages.
+
+_SPARK_SERVICE_REGISTRY = {}
+
+
+def GetSparkServiceClass(cloud, service_type):
+  """Get the Spark class corresponding to 'cloud'."""
+  if service_type == PKB_MANAGED:
+    return _SPARK_SERVICE_REGISTRY.get(service_type)
+  elif cloud in _SPARK_SERVICE_REGISTRY:
+    return _SPARK_SERVICE_REGISTRY.get(cloud)
+  else:
+    raise Exception('No Spark service found for {0}'.format(cloud))
+
+
+class AutoRegisterSparkServiceMeta(abc.ABCMeta):
+  """Metaclass which allows SparkServices to register"""
```

Comment:
crazy nit: would you mind ending the first line of docstrings with punctuation? I noticed this in a couple other places